Compilation error when including `yaml-cpp/node/detail/iterator.h`

The following program does not compile:

```
#include <yaml-cpp/node/detail/iterator.h>
#include <yaml-cpp/node/node.h>

int main() {
}
```

The error with GCC 6.2.0 is the following:

```
In file included from test.cpp:1:0:
include/yaml-cpp/node/detail/iterator.h: In member function 'YAML::detail::iterator_base<V>::value_type YAML::detail::iterator_base<V>::operator*() const':
include/yaml-cpp/node/detail/iterator.h:76:43: error: there are no arguments to 'Node' that depend on a template parameter, so a declaration of 'Node' must be available [-fpermissive]
       return value_type(Node(*v, m_pMemory));
                                           ^
include/yaml-cpp/node/detail/iterator.h:76:43: note: (if you use '-fpermissive', G++ will accept your code, but allowing the use of an undeclared name is deprecated)
include/yaml-cpp/node/detail/iterator.h:78:49: error: there are no arguments to 'Node' that depend on a template parameter, so a declaration of 'Node' must be available [-fpermissive]
       return value_type(Node(*v.first, m_pMemory), Node(*v.second, m_pMemory));
                                                 ^
include/yaml-cpp/node/detail/iterator.h:78:77: error: there are no arguments to 'Node' that depend on a template parameter, so a declaration of 'Node' must be available [-fpermissive]
       return value_type(Node(*v.first, m_pMemory), Node(*v.second, m_pMemory));
                                                                             ^
```

But it compiles fine when reversing the order of the include directives:
```
#include <yaml-cpp/node/node.h>
#include <yaml-cpp/node/detail/iterator.h>

int main() {
}
```

--- a/include/yaml-cpp/node/detail/iterator.h
+++ b/include/yaml-cpp/node/detail/iterator.h
@@ -8,6 +8,7 @@
 #endif
 
 #include "yaml-cpp/dll.h"
+#include "yaml-cpp/node/node.h"
 #include "yaml-cpp/node/ptr.h"
 #include "yaml-cpp/node/detail/node_iterator.h"
 #include <cstddef>

